docs/source/app-dev/app-arch.rst

```diff
@@ -1,295 +1,120 @@
+If you choose a different frontend framework, the DAML assistant will also help you generate code
+for a variety of other programming languages. Run ``daml codegen --help`` to find out more.
 
-In particular, all applications need to handle the asynchronous nature of the ledger API. The most important consequence of this is that applications must be multi-threaded. This is because of the asynchronous, separate streams of commands, transaction and completion events.
+Backend
+~~~~~~~
 
-Although you can choose to do this in several ways, from bare threads (such as a Java Thread) through thread libraries, generally the most effective way of handling this is by adopting a reactive architecture, often using a library such as `RxJava <https://github.com/ReactiveX/RxJava>`__.
+The backend for your application can be any DAML ledger implementation. For applications where
+performance is not of outmost importance, we recommend using the :ref:`DAML JSON API <json-api>` as
+an interface. It is served by the HTTP JSON API server connected to the ledger API server. It
+provides simple HTTP endpoints to interact with the ledger via GET/POST requests. If performance
+does become a concern for you application, you can drop the HTTP JSON API server and use the
+:ref:`gRPC API <grpc>` directly.
 
-All the language bindings support this reactive pattern as a fundamental requirement.
+When you use the ``create-daml-app`` template application you can start a local sandbox together
+with a JSON API server by running
 
-.. _event-driven-applications-1:
+.. code-block:: bash
 
-Structuring event-driven applications
-=====================================
+  ./daml-start.sh
 
-Event-driven applications read a stream of transaction events from the ledger, and convert them to some other representation. This may be a record on a database, some update of a UI, or a differently formatted message that is sent to an upstream process. It may also be a command that transforms the ledger.
+in the root of the project. This is the most simple DAML ledger implementation. Once your
+application matures and becomes ready for production, the ``daml deploy`` command helps you deploy
+your frontend and DAML artefacts of your project to a production ledger. See :ref:`Deploying to DAML
+Ledgers <deploy-ref_index>` for an in depth manual for specific ledgers.
 
-The critical thing here is that each event is processed in isolation - the application does not need to keep any application-related state between each event. It is this that differentiates it from a state-driven application.
+Authentication
+~~~~~~~~~~~~~~
 
-To do this, the application should:
+DAML ledgers support a unified interface for authentication of commands, a detailed description can
```

Comment:
Add a sentence why authentication is needed at the beginning of the section, e.g.,

To run your application against a :ref: `deployed ledger <deploy-ref_overview>`, you will need to add authentication.

Reply:
done.